Components/StatsDisplay/index.js

```diff
@@ -19,12 +35,58 @@ export default function StatsDisplay({
   selectedSeason,
   selectedSeasonTwo,
 }) {
+  const [barSelection, setBarSelection] = useState("pts");
+  const [label, setLabel] = useState("pts");
+
   const URL = "https://www.balldontlie.io/api/v1/players/";
 
   const { data: player } = useSWR(URL + selectedPlayer);
-
   const { data: playerTwo } = useSWR(URL + selectedPlayerTwo);
 
+  let statsArray = [];
+  if (playerStats && playerTwoStats) {
+    statsArray = [{ ...playerStats }, { ...playerTwoStats }];
+  }
+
+  const [chartData, setChartData] = useState({
+    labels: [
+      `${player?.first_name ?? "Player 1"} ${player?.last_name ?? ""}`,
+      `${playerTwo?.first_name ?? "Player 2"} ${playerTwo?.last_name ?? ""}`,
+    ],
+    datasets: [
+      {
+        label: label,
+        data: statsArray.map((data) => data[barSelection]),
+      },
+    ],
+  });
+
+  useEffect(() => {
+    setChartData((prevChartData) => ({
+      ...prevChartData,
+      labels: [
+        `${player?.first_name ?? "Player 1"} ${player?.last_name ?? ""}`,
+        `${playerTwo?.first_name ?? "Player 2"} ${playerTwo?.last_name ?? ""}`,
+      ],
+      datasets: [
+        {
+          label: label,
+          data: statsArray.map((data) => data[barSelection]),
+          backgroundColor: ["#6E941B", "#442594"],
+        },
+      ],
+    }));
+  }, [
+    player,
+    playerTwo,
+    playerStats,
+    playerTwoStats,
+    selectedPlayer,
+    selectedPlayerTwo,
+    selectedSeason,
+    selectedSeasonTwo,
+    barSelection,
+  ]);
```

Comment:
This is the `useEffect` police - you can avoid this effect (and the chartData state) by using a `const`. This works because the component re-renders every time any of the values you use in constructing this object changes.

```suggestion
  const chartData = {
    labels: [
      `${player?.first_name ?? "Player 1"} ${player?.last_name ?? ""}`,
      `${playerTwo?.first_name ?? "Player 2"} ${playerTwo?.last_name ?? ""}`,
    ],
    datasets: [
      {
        label: label,
        data: statsArray.map((data) => data[barSelection]),
        backgroundColor: ["#6E941B", "#442594"],
      },
    ],
  };
```

Reply:
This absolutely makes sense. Guess I have learned a lot about useEffect and states this week. Thanks!